molgenis-hpofilter/src/main/java/org/molgenis/hpofilter/HpoFilterController.java

```diff
@@ -0,0 +1,270 @@
+package org.molgenis.hpofilter;
+
+import static org.molgenis.hpofilter.HpoFilterController.URI;
+import static org.molgenis.security.core.utils.SecurityUtils.AUTHORITY_ENTITY_READ_PREFIX;
+import static org.molgenis.security.core.utils.SecurityUtils.AUTHORITY_SU;
+import static org.molgenis.security.core.utils.SecurityUtils.currentUserHasRole;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+
+import org.apache.commons.lang3.StringUtils;
+import org.molgenis.MolgenisFieldTypes;
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.DataService;
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.Repository;
+import org.molgenis.data.RepositoryDecoratorFactory;
+import org.molgenis.data.support.DefaultAttributeMetaData;
+import org.molgenis.data.support.DefaultEntityMetaData;
+import org.molgenis.data.support.MapEntity;
+import org.molgenis.framework.ui.MolgenisPluginController;
+import org.molgenis.hpofilter.data.GeneMapProvider;
+import org.molgenis.hpofilter.data.HGNCLocations;
+import org.molgenis.hpofilter.data.HpoFilterDataProvider;
+import org.molgenis.hpofilter.data.Locus;
+import org.molgenis.hpofilter.utils.HgncLocationsUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import com.google.common.collect.Lists;
+
+@Controller
+@RequestMapping(URI)
+public class HpoFilterController extends MolgenisPluginController
```

Comment:
Use a Logger to provide debug or info instead of system prints. Most controllers have an example.